File: comgongdaeng/TwoPointer/BOJ_1644.py
import sys
N = int(sys.stdin.readline())

# 에라토스테네스의 체
# 2, 3, 5, 7의 배수를 제거
a = [False, False] + [True] * (N-1)
prime_num = []

for i in range(2, N + 1):
    if a[i]:
        prime_num.append(i)
        for j in range(2*i, N+1, i):
            a[j] = False

answer = 0
left = 0
right = 0
while right <= len(prime_num):
    temp_sum = sum(prime_num[left:right])
    if temp_sum == N:
        answer += 1
        right += 1
    elif temp_sum < N:
        right += 1
    else:
        left += 1

print(answer)

File: keemsebin/TwoPointer/BOJ_1644.js
const readline = require("readline");
const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout,
});

let input = [];
let arr = [];

rl.on("line", function (line) {
  input.push(line);
}).on("close", function () {
  let N = Number(input[0]);
  let prime = Array(N + 1).fill(true);
  prime[0] = prime[1] = false;

  for (let i = 2; i <= N; i++) {
    // 소수면 true, 아니면 false
    if (prime[i]) {
      for (let j = i * i; j <= N; j += i) {
        prime[j] = false;
      }
    }
  }
  for (let i = 0; i <= N; i++) {
    if (prime[i]) arr.push(i);
  }

  let left = 0;
  let right = 0;
  let sum = 0;
  let ans = 0;

  while (true) {
    if (sum >= N) {
      if (sum === N) ans++;
      sum -= arr[left++];
    } else if (right < arr.length) {
      sum += arr[right++];
    } else {
      break;
    }
  }
  console.log(ans);
});
